server/graphql/v2/input/OrderDetailsInput.ts

```diff
@@ -0,0 +1,19 @@
+import { GraphQLInputObjectType } from 'graphql';
+
+import { AmountInput } from './AmountInput';
+
+export const OrderDetailsInput = new GraphQLInputObjectType({
+  name: 'OrderDetailsInput',
+  description: 'Input to set amount details received by the host',
+  fields: () => ({
+    totalAmount: {
+      type: AmountInput,
+    },
```

Comment:
We're not exposing `totalAmount` so far.

In `OrderCreateInput`, we have `amount`

> description: 'The contribution amount for 1 quantity, without platform contribution and taxes',

I think it's best to re-use this definition, so we don't introduce yet a different definition with a different name.

This mean that frontend will have to construct somehow it by doing `amount = amountReceived + paymentProcessorFeesAmount - platformTipAmount`.

And that you will have to reconstruct it `totalAmount = amount + platformTipAmount`

Reply:
True, I think `totalAmount` is coercing the frontend to speak backend abstractions.

Following what was designed, we want:

![Screenshot from 2021-12-16 10-51-44](https://user-images.githubusercontent.com/2119706/146384385-21814dcd-fde8-4fa0-8e24-72bd291fe93f.png)

"Amount received by Accountable": Is this the amount received by the host (donation + tip)?
"Payment processor fees": This is the difference between the amount initiated and the amount received.
"Platform tip amount": Intended tip. To deduct from the amount received.

Is that correct?

Reply:
I think so.

I added some comments there:
https://github.com/opencollective/opencollective-frontend/pull/7268#discussion_r770545634